ironfish/src/rpc/routes/chain/getTransactionStream.ts

```diff
@@ -106,28 +143,55 @@ router.register<typeof GetTransactionStreamRequestSchema, GetTransactionStreamRe
       head: head,
     })
 
-    const processBlock = (block: Block, type: 'connected' | 'disconnected' | 'fork'): void => {
+    const processBlock = async (
+      block: Block,
+      type: 'connected' | 'disconnected' | 'fork',
+    ): Promise<void> => {
       const transactions: Transaction[] = []
 
       for (const tx of block.transactions) {
         const notes = new Array<Note>()
+        const mints = new Array<Mint>()
+        const burns = new Array<Burn>()
 
         for (const note of tx.notes) {
           const decryptedNote = note.decryptNoteForOwner(request.data.incomingViewKey)
 
           if (decryptedNote) {
+            const assetValue = await validatedAssetValue(decryptedNote.assetIdentifier())
             notes.push({
-              amount: CurrencyUtils.encode(decryptedNote.value()),
+              value: CurrencyUtils.encode(decryptedNote.value()),
               memo: decryptedNote.memo(),
+              assetId: decryptedNote.assetIdentifier.toString(),
+              assetName: assetValue.name.toString(),
             })
           }
         }
 
-        if (notes.length) {
+        for (const burn of tx.burns) {
+          const assetValue = await validatedAssetValue(burn.assetIdentifier)
+          burns.push({
+            value: CurrencyUtils.encode(burn.value),
+            assetId: burn.assetIdentifier.toString('hex'),
+            assetName: assetValue.name.toString(),
+          })
+        }
+
+        for (const mint of tx.mints) {
+          mints.push({
+            value: CurrencyUtils.encode(mint.value),
+            assetId: mint.asset.identifier().toString('hex'),
+            assetName: mint.asset.name().toString(),
+          })
+        }
+
+        if (notes.length || burns.length || mints.length) {
```

Comment:
Is this conditional what we want? Just verifying, it was 
`if (notes.length) {`
before. Now with possibility of burns/mints I updated to this